content/en/blog/_posts/2025-04-23-Kubernetes-v1-33-Release/index.md

```diff
@@ -0,0 +1,471 @@
+
+This support will not impact existing Pods unless you manually specify `pod.spec.hostUsers` to opt in. As highlighted in the [v1.30 sneak peek blog](/blog/2024/03/12/kubernetes-1-30-upcoming-changes/), this is an important milestone for mitigating vulnerabilities.
+
+This work was done as part of [KEP-127: Support User Namespaces in pods](https://kep.k8s.io/127) led by SIG Node.
+
+### Pod `procMount` option
+
+The `procMount` option, introduced as alpha in v1.12, and off-by-default beta in v1.31, has moved to an on-by-default beta in v1.33. This enhancement improves Pod isolation by allowing users to fine-tune access to the `/proc` filesystem. Specifically, it adds a field to the Pod `securityContext` that lets you override the default behavior of masking and marking certain `/proc` paths as read-only. This is particularly useful for scenarios where users want to run unprivileged containers inside the Kubernetes Pod using user namespaces. Normally, the container runtime (via the CRI implementation) starts the outer container with strict `/proc` mount settings. However, to successfully run nested containers with an unprivileged Pod, users need a mechanism to relax those defaults, and this feature provides exactly that.
+
+This work was done as part of [KEP-4265: add ProcMount option](https://kep.k8s.io/4265) led by SIG Node.
+
+### CPUManager policy to distribute CPUs across NUMA nodes
+
+This feature adds a new policy option for the CPU Manager to distribute CPUs across Non-Uniform Memory Access (NUMA) nodes, rather than concentrating them on a single node. It optimizes CPU resource allocation by balancing workloads across multiple NUMA nodes, thereby improving performance and resource utilization in multi-NUMA systems.
+
+This work was done as part of [KEP-2902: Add CPUManager policy option to distribute CPUs across NUMA nodes instead of packing them](https://kep.k8s.io/2902) led by SIG Node.
+
+### Zero-second sleeps for container PreStop hooks
+
+Kubernetes 1.29 introduced a Sleep action for the `preStop` lifecycle hook in Pods, allowing containers to pause for a specified duration before termination.
+This provides a straightforward method to delay container shutdown, facilitating tasks such as connection draining or cleanup operations.
+
+The Sleep action in a `preStop` hook can now accept a zero-second duration as a beta feature. This allows defining a no-op `preStop` hook, which is useful when a `preStop` hook is required but no delay is desired.
+
+This work was done as part of [KEP-3960: Introducing Sleep Action for PreStop Hook](https://kep.k8s.io/3960) and [KEP-4818: Allow zero value for Sleep Action of PreStop Hook](https://kep.k8s.io/4818) led by SIG Node.
+
+### Declarative validation of Kubernetes-native types
```

Comment:
Who is the audience for this blog? While this feature is certainly impactful and helpful to Kubernetes *maintainers*, I'm not sure whether this will be interesting to Kubernetes *consumers*.

Reply:
Technically the audience could be many -- but surely the main target is Kubernetes users / consumers. I do think having a few items targeting maintainers (or future maintainers) is fine, but this could probably use some title update to clarify that. I adjusted a little with 006509a4c1

 This was originally [highlighted from SIG API Machinery](https://github.com/kubernetes/sig-release/discussions/2734#discussioncomment-12632888), and was then questioned whether this is worth highlighting or not. If this seems to be too much of implementation detail, we could drop this section entirely.
CC'ing @deads2k and @jpbetz for their input as well 🙏

Reply:
As I can see this highlight being an important part of the release, I'm resolving this (and there is nothing wrong about having one more section, as we already have 30+ highlights)